doc/deployment-schema.json

```diff
@@ -0,0 +1,310 @@
+                                "routes": {
+                                    "type": "object",
+                                    "patternProperties": {
+                                        "^.+$": {
+                                            "type": "string",
+                                            "examples": [
+                                                "FROM /* INTO $upstream"
+                                            ]
+                                        }
+                                    }
+                                },
+                                "storeAndForwardConfiguration": {
+                                    "type": "object",
+                                    "required": [
+                                        "timeToLiveSecs"
+                                    ],
+                                    "properties": {
+                                        "timeToLiveSecs": {
+                                            "type": "integer",
+                                            "examples": [
+                                                7200
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "patternProperties": {
+                "^.+$": {
+                    "type": "object",
+                    "required": [
+                        "properties.desired"
+                    ],
+                    "properties": {
+                        "properties.desired": {
+                            "type": "object"
+                        }
+                    }
+                }
+            },
+            "additionalProperties": false
+        }
+    },
+    "definitions": {
+        "moduleType": {
+            "enum": [
+                "docker"
+            ]
+        },
+        "status": {
+            "enum": [
+                "running",
+                "stopped"
+            ]
+        },
+        "restartPolicy": {
+            "enum": [
+                "never",
+                "on-failure",
+                "on-unhealthy",
+                "always"
+            ]
+        },
+        "moduleSettings": {
+            "type": "object",
+            "required": [
+                "image"
+            ],
+            "properties": {
+                "image": {
+                    "type": "string",
+                    "examples": [
+                        "mcr.microsoft.com/azureiotedge-agent:1.0"
+                    ]
+                },
+                "createOptions": {
+                    "$ref": "#/definitions/createOptions"
+                }
+            }
+        },
+        "env": {
+            "type": "object",
+            "patternProperties": {
+                "^.+$": {
```

Comment:
Do we allow space?

Reply:
I changed the pattern, space should not be allowed